src/message-backup/BackupClient.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Where message backups should be stored
+ */
+export enum BackupProvider {
+  none,
+  xmtp,
+}
+export type SelectBackupProvider = () => Promise<BackupProvider>
+
+export interface BackupConfiguration {
+  provider: BackupProvider
+  // For the XMTP BackupProvider, this is the name of a topic
+  location?: string
+  // For the XMTP BackupProvider, this is the symmetric encryption key used to encrypt/decrypt backups
+  encryptionKey?: string
```

Comment:
Why string? Using strings for keys usually makes me suspicious as they bring in ambiguity about how exactly will they be turned into key bytes.